File: st_notebook/bixecamp/aula_2/7.cpp
// Notes

// If you have the input "1234" and i = 0, this means that the string "1234" will be read and each character
// will be assigned to m[0][0], m[0][1], m[0][2], and m[0][3] in a char matrix.

// Using ASCII, for digits from 0 to 9, it is possible to convert them to int by simply subtracting
// the char '0'. https://pt.stackoverflow.com/questions/3760/o-que-acontece-em-uma-convers%C3%A3o-de-um-char-para-um-int

#include <bits/stdc++.h>
using namespace std;

const int N = 4;

int main() {

    char m[N][N];

    int k;
    cin >> k;
    bool able = true;
    k = k*2;
    int count[10] = { 0 };

    for (int i = 0; i < N; i ++) {
        cin >> m[i]; 
        for (int j = 0; j < N; j++) {
            if (m[i][j] != '.') {
                count[m[i][j] - '0']++;
            }
        }
    }

    for (int i = 0; i < 10; i++) {
        if (count[i] > k) {
            able = false;
        }
    }

    if (able) {
        cout << "YES" << "\n";
    } else {
        cout << "NO" << "\n";
    }

    return 0;
}

File: st_notebook/bixecamp/aula_2/7.py
# this shit got AC ok

n = int(input())
n = n*2

l1 = list(input()) 
l2 = list(input())
l3 = list(input())
l4 = list(input())
able = True

for k in range(4):
    if l1[k] != '.':
        l1[k] = int(l1[k])
    if l2[k] != '.':
        l2[k] = int(l2[k])
    if l3[k] != '.':
        l3[k] = int(l3[k])
    if l4[k] != '.':
        l4[k] = int(l4[k])
            
for i in range(1, 10):
    cnt = 0
            
    for k in range(4):
        if (l1[k] == i):
            cnt += 1
        if (l2[k] == i):
            cnt += 1
        if (l3[k] == i):
            cnt += 1
        if (l4[k] == i):
            cnt += 1
    if cnt > n:
        able = False
        break

if able:
    print('YES')
else:
    print('NO')